src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -57,56 +62,35 @@ var kubeconfigCmd = &cobra.Command{
 		cmd.SilenceUsage = true
 
 		configDir := path.Join(homedir.HomeDir(), ".kube")
-		linkPath := path.Join(configDir, "config")
-		if show {
-			// The output is "true", "false", or an error message for UI.
-			// We will only return nil in this path.
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					fmt.Println("false")
-				} else if errors.Is(err, syscall.EINVAL) {
-					fmt.Printf("File %s exists and is not a symlink\n", linkPath)
-				} else {
-					fmt.Printf("%s\n", err)
-				}
-			} else if target == configPath {
-				fmt.Println("true")
-			} else {
-				// For a symlink pointing elsewhere, we assume we can overwrite.
-				fmt.Println("false")
-			}
-			return nil
+
+		configFile, err := os.Open(configPath)
+		if err != nil {
+			return err
 		}
+
+		kubeConfig, err := updateClusterIP(configFile, rdNetworking)
+
+		var finalKubeConfigFile *os.File
 		if enable {
+			finalKubeConfigFile, err = os.Create(configDir)
+			if err != nil {
+				return err
+			}
+			defer finalKubeConfigFile.Close()
 			err = os.Mkdir(configDir, 0o750)
 			if err != nil && !errors.Is(err, os.ErrExist) {
 				// The error already contains the full path, we can't do better.
 				return err
 			}
-			err = os.Symlink(configPath, linkPath)
+			err = yaml.NewEncoder(finalKubeConfigFile).Encode(kubeConfig)
 			if err != nil {
-				if errors.Is(err, os.ErrExist) {
-					// If it already exists, do nothing; even if it's not a symlink.
-					return nil
-				}
 				return err
 			}
 		} else {
-			// No need to create if we want to remove it
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					return nil
-				}
+			err = os.Remove(finalKubeConfigFile.Name())
```

Comment:
This is clobbering _any_ kubeconfig file the user might have, not just if we wrote it… That would be wrong if they're (for some reason) talking to EKS or something.